packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp

```diff
@@ -7576,6 +7609,272 @@ namespace Tpetra {
+                            const Kokkos::DualView<const local_ordinal_type*, buffer_device_type>& permuteFromLIDs,
+                            LocalOrdinal loopEnd)
+  {
+    using crs_graph_type = CrsGraph<LocalOrdinal, GlobalOrdinal, Node>;
+    using LO = LocalOrdinal;
+    using GO = GlobalOrdinal;
+    typedef typename crs_graph_type::global_inds_device_view_type::non_const_value_type global_inds_device_value_t;
+    typedef typename crs_graph_type::local_graph_device_type k_local_graph_device_type;
+    typedef typename Node::execution_space exec_space;
+    typedef Kokkos::RangePolicy<exec_space, LO> range_type;
+
+    const LocalOrdinal LINV = Teuchos::OrdinalTraits<LocalOrdinal>::invalid ();
+    const GlobalOrdinal GINV = Teuchos::OrdinalTraits<GlobalOrdinal>::invalid ();
+
+    const k_local_graph_device_type & srcGraphDevice = srcCrsGraph.getLocalGraphDevice();
+    const k_local_graph_device_type & tgtGraphDevice = tgtCrsGraph.getLocalGraphDevice();
+
+    using local_map_type = typename crs_graph_type::map_type::local_map_type;
+    local_map_type srcRowMapLocal = srcCrsGraph.getRowMap()->getLocalMap();
+    local_map_type srcColMapLocal = srcCrsGraph.getColMap()->getLocalMap();
+    local_map_type tgtRowMapLocal = tgtCrsGraph.getRowMap()->getLocalMap();
+
+    auto tgtLocalRowPtrsDevice = tgtCrsGraph.getRowPtrsUnpackedDevice();
+    auto tgtGlobalColInds = tgtCrsGraph.gblInds_wdv.getDeviceView(Access::ReadWrite);
+    auto srcLocalRowPtrsDevice = srcCrsGraph.getLocalRowPtrsDevice();
+    auto srcLocalColIndsDevice = srcCrsGraph.lclIndsUnpacked_wdv.getDeviceView(Access::ReadOnly);
+
+    typename crs_graph_type::num_row_entries_type::non_const_type h_numRowEnt = tgtCrsGraph.k_numRowEntries_;
+
+    auto k_numRowEnt = Kokkos::create_mirror_view_and_copy (device_type (), h_numRowEnt);
+
+    const bool sorted = false;
+
+    bool hasMap = permuteFromLIDs.extent(0) > 0;
+    auto permuteToLIDs_d = permuteToLIDs.view_device ();
+    auto permuteFromLIDs_d = permuteFromLIDs.view_device ();
+
+#ifdef CRSGRAPH_INNER_ABORT
+#undef CRSGRAPH_INNER_ABORT
+#endif
+
+#define CRSGRAPH_INNER_ABORT(lin) do {            \
+      printf("ERROR: Tpetra_CrsGraph_def.hpp:%d", lin);           \
+      Kokkos::abort("error");                   \
+    } while(0)
+
+    Kokkos::parallel_for("Tpetra_CrsGraph::copyAndPermuteNew2",
+                         range_type (0, loopEnd),
+                         KOKKOS_LAMBDA(const LO sourceLID)
+                         {
+                           auto srcLid = sourceLID;
+                           auto tgtLid = sourceLID;
+                           if (hasMap) {
+                             srcLid = permuteFromLIDs_d(srcLid);
+                             tgtLid = permuteToLIDs_d(tgtLid);
+                           }
+                           auto srcGid = srcRowMapLocal.getGlobalElement(srcLid);
+                           if (srcGid == GINV) CRSGRAPH_INNER_ABORT(__LINE__);
```

Comment:
Can this and the following that call `CRSGRAPH_INNER_ABORT` happen, or is this just leftover debugging code?